src/common/scheduler/scheduler_store.go

```diff
@@ -0,0 +1,140 @@
+package scheduler
+
+import "sync"
+import "strings"
+
+const defaultSize = 10
+
+//Store define the basic operations for storing and managing policy watcher.
+//The concrete implementation should consider concurrent supporting scenario.
+//
+type Store interface {
+	//Put a new policy in.
+	Put(key string, value *Watcher)
+
+	//Get the corresponding policy with the key.
+	Get(key string) *Watcher
+
+	//Check if the key existing in the store.
+	Exists(key string) bool
+
+	//Remove the specified policy and return its reference.
+	Remove(key string) *Watcher
+
+	//Size return the total count of items in store.
+	Size() uint32
+
+	//Get all the items in the store.
+	GetAll() []*Watcher
+
+	//Clear store.
+	Clear()
+}
+
+//ConcurrentStore implements Store interface and supports concurrent operations.
+type ConcurrentStore struct {
+	//Read-write mutex to synchronize the data map.
+	mutex *sync.RWMutex
```

Comment:
Use a anonymous " *sync.RWMutex" may be more idiomatic, so you only need 
```
cs.Lock()....
```
instead of cs.mutex.Lock()